plugins/pluto/src/ad936xadvanced.cpp

```diff
@@ -0,0 +1,198 @@
+
+	QPushButton *txMonitorBtn = new QPushButton("TX MONITOR", this);
+	Style::setStyle(txMonitorBtn, style::properties::button::blueGrayButton);
+	txMonitorBtn->setCheckable(true);
+
+	TxMonitorWidget *txMonitor = new TxMonitorWidget(m_uri, centralWidget);
+	centralWidget->addWidget(txMonitor);
+
+	connect(this, &AD936XAdvanced::readRequested, txMonitor, &TxMonitorWidget::readRequested);
+
+	connect(txMonitorBtn, &QPushButton::clicked, this, [=, this]() { centralWidget->setCurrentWidget(txMonitor); });
+
+	QPushButton *auxAdcDacIioBtn = new QPushButton("Aux ADC/DAC/IIO", this);
+	Style::setStyle(auxAdcDacIioBtn, style::properties::button::blueGrayButton);
+	auxAdcDacIioBtn->setCheckable(true);
+
+	AuxAdcDacIoWidget *auxAdcDacIo = new AuxAdcDacIoWidget(m_uri, centralWidget);
+	centralWidget->addWidget(auxAdcDacIo);
+
+	connect(this, &AD936XAdvanced::readRequested, auxAdcDacIo, &AuxAdcDacIoWidget::readRequested);
+
+	connect(auxAdcDacIioBtn, &QPushButton::clicked, this,
+		[=, this]() { centralWidget->setCurrentWidget(auxAdcDacIo); });
+
+	QPushButton *miscBtn = new QPushButton("MISC", this);
+	Style::setStyle(miscBtn, style::properties::button::blueGrayButton);
+	miscBtn->setCheckable(true);
+
+	MiscWidget *misc = new MiscWidget(m_uri, centralWidget);
+	centralWidget->addWidget(misc);
+
+	connect(this, &AD936XAdvanced::readRequested, misc, &MiscWidget::readRequested);
+
+	connect(miscBtn, &QPushButton::clicked, this, [=, this]() { centralWidget->setCurrentWidget(misc); });
+
+	QPushButton *bistBtn = new QPushButton("BIST", this);
+	Style::setStyle(bistBtn, style::properties::button::blueGrayButton);
+	bistBtn->setCheckable(true);
+
+	BistWidget *bist = new BistWidget(m_uri, centralWidget);
+	centralWidget->addWidget(bist);
+
+	connect(this, &AD936XAdvanced::readRequested, bist, &BistWidget::readRequested);
+
+	connect(bistBtn, &QPushButton::clicked, this, [=, this]() { centralWidget->setCurrentWidget(bist); });
+
+	navigationButtons->addButton(ensmModeClocksBtn);
+	navigationButtons->addButton(eLnaBtn);
+	navigationButtons->addButton(rssiBtn);
+	navigationButtons->addButton(gainBtn);
+	navigationButtons->addButton(txMonitorBtn);
+	navigationButtons->addButton(auxAdcDacIioBtn);
+	navigationButtons->addButton(miscBtn);
+	navigationButtons->addButton(bistBtn);
+
+	m_tool->addWidgetToTopContainerHelper(ensmModeClocksBtn, TTA_LEFT);
+	m_tool->addWidgetToTopContainerHelper(eLnaBtn, TTA_LEFT);
+	m_tool->addWidgetToTopContainerHelper(rssiBtn, TTA_LEFT);
+	m_tool->addWidgetToTopContainerHelper(gainBtn, TTA_LEFT);
+	m_tool->addWidgetToTopContainerHelper(txMonitorBtn, TTA_LEFT);
+	m_tool->addWidgetToTopContainerHelper(auxAdcDacIioBtn, TTA_LEFT);
+	m_tool->addWidgetToTopContainerHelper(miscBtn, TTA_LEFT);
+	m_tool->addWidgetToTopContainerHelper(bistBtn, TTA_LEFT);
+
+	// Style::setStyle(scrollWidget, style::properties::widget::border_interactive, true, true);
+}
+
+AD936XAdvanced::~AD936XAdvanced()
+{
+	// close Connection
+	ConnectionProvider::close(m_uri);
```

Comment:
similar to what i described above, you close an instance here, but no instance was opened in this class